routes/dao_coin_exchange.go

```diff
@@ -74,16 +82,115 @@ func (fes *APIServer) GetDAOCoinLimitOrders(ww http.ResponseWriter, req *http.Re
 		return
 	}
 
+	txnStatus := requestData.TxnStatus
+	if txnStatus == "" {
+		txnStatus = TxnStatusUnconfirmed
+	} else if txnStatus != TxnStatusUnconfirmed &&
+		txnStatus != TxnStatusConfirmed {
+
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("GetDAOCoinLimitOrders: Invalid TxnStatus: %v. Options "+
+				"are {unconfirmed, confirmed}.", txnStatus),
+		)
+		return
+	}
+
 	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
 	if err != nil {
 		_AddInternalServerError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Problem fetching utxoView: %v", err))
 		return
 	}
+	if txnStatus == TxnStatusConfirmed {
+		utxoView = lib.NewUtxoView(
+			fes.backendServer.GetBlockchain().DB(),
+			fes.Params,
+			nil,
+			fes.backendServer.GetBlockchain().Snapshot(),
+			nil)
+	}
+
+	// If they requested a specific set of OrderIds then fetch them directly
+	if len(requestData.OrderIds) > 0 {
+		// Don't allow fetching more than this many orders
+		maxOrdersToFetch := 10000
+		if len(requestData.OrderIds) > maxOrdersToFetch {
+			_AddBadRequestError(
+				ww,
+				fmt.Sprintf("GetDAOCoinLimitOrders: Cannot fetch more than %v orders at once", maxOrdersToFetch),
+			)
+			return
+		}
+		ordersToReturn := make([]DAOCoinLimitOrderEntryResponse, 0, len(requestData.OrderIds))
+		for _, orderIdHex := range requestData.OrderIds {
+			orderIdBytes, err := hex.DecodeString(orderIdHex)
+			if err != nil {
+				_AddBadRequestError(
+					ww,
+					fmt.Sprintf("GetDAOCoinLimitOrders: Invalid OrderId: %v", err),
+				)
+				return
+			}
+			if len(orderIdBytes) != 32 {
+				_AddBadRequestError(
+					ww,
+					fmt.Sprintf("GetDAOCoinLimitOrders: Block hash has length (%d) but should "+
+						"be (%d)", len(orderIdBytes), 32),
+				)
+				return
+			}
+			orderId := lib.BlockHash{}
+			copy(orderId[:], orderIdBytes)
+			orderEntry, err := utxoView.GetDAOCoinLimitOrderEntry(&orderId)
+			if err != nil {
+				_AddBadRequestError(
+					ww,
+					fmt.Sprintf("GetDAOCoinLimitOrders: Error fetching order: %v", err),
+				)
+				return
+			}
+			if orderEntry == nil {
+				continue
+			}
+			if orderEntry.IsDeleted() {
+				continue
+			}
+			orderRes, err := buildDAOCoinLimitOrderResponse(
+				lib.Base58CheckEncode(orderEntry.TransactorPKID[:], false, fes.Params),
```

Comment:
We need to look up the public keys for these PKIDs. If something had a swap identity, this would lead to unexpected behavior.